src/Type/Php/RegexExpressionHelper.php

```diff
@@ -44,7 +44,7 @@ public function resolve(Expr $expr): Type
 					&& $expr->name instanceof Name
 					&& $expr->name->toLowerString() === 'preg_quote'
 				) {
-					return new ConstantStringType('');
+					return new ConstantStringType('.*');
```

Comment:
instead of dummy-replace `preg_quote($x)` with `''` we use `.*` as a placeholder, so we don't produce invalid patterns

Reply:
Can you please explain the constant string meaning? AFAICT such assumption here can produce misleading assumptions/phpstan errors.

Reply:
We replace a preg_quote with some constant value so we can finally resolve a string concatenation to a pattern at analysis time. 

The preg_quoted value cannot return something which creates capturing groups or syntax errors, therefore we simplify it to improve coverage

Reply:
It'd probably help to have some more comments explaining these edge cases / hacks to future archeologists 😅 but PR looks good thanks